### PR TITLE
Split screen case search "search again" workflow

### PR DESCRIPTION
**Pull request by nospame**
## Product Description
For all apps under domains with the split screen case search feature flag enabled, hides "search again" options from the app manager UI and removes it from the app build.

## Technical Summary
[USH-3178](https://dimagi-dev.atlassian.net/browse/USH-3178)
During app build, checks if the split screen case search feature flag is enabled when adding search actions. If it is, and the action is on the case search screen (i.e., is a "search again" action), that action is not added to the app's xml. Also adds a feature flag check to hide the app manager ui components for "search again" label/icon/audio.

## Feature Flag
split_screen_case_search

### Safety story
Will only affect domains with the feature flag enabled (none yet on production), but will apply to all apps under that domain. As with similar flagged options in app manager, leaves any previous data stored in the model rather than clearing it out.

### Automated test coverage
Added automated test.

### QA Plan
QA planned as part of split screen case search overall.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-3178]: https://dimagi-dev.atlassian.net/browse/USH-3178?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_suite.py [hunk 1]
```diff
@@ -77,6 +79,46 @@ def test_case_search_action(self):
             "./detail[@id='m0_search_short']/action"
         )
 
+    @flag_enabled('SPLIT_SCREEN_CASE_SEARCH')
+    def test_splitscreen_case_search_removes_search_again(self):
```
Comment (nospame) [inline corehq/apps/app_manager/tests/test_suite.py:83]: Thinking there might be a better place for this test but I'm not sure.

File: corehq/apps/app_manager/suite_xml/sections/details.py [hunk 2]
```diff
@@ -235,7 +235,9 @@ def build_detail(self, module, detail_type, detail, detail_column_infos, tabs=No
 
                 if module_offers_search(module) and not module_uses_inline_search(module):
                     in_search = module_loads_registry_case(module) or "search" in id
-                    d.actions.append(self._get_case_search_action(module, in_search=in_search))
+                    # don't add search again action in split screen
```
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/sections/details.py:238]: Heads up that https://github.com/dimagi/commcare-hq/pull/32974 is going to conflict/affect this. Can you merge master when you have a chance?

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 3]
```diff
@@ -155,23 +155,25 @@ <h4 class="panel-title panel-title-nolink">{% trans "Search and Claim Options" %
...
           </div>
-          <div class="form-group" data-bind="hidden: inlineSearchActive">
-            <label for="search_again_label" class="{% css_label_class %} control-label">
-              {% trans "Label for Searching Again" %}
-              <span class="hq-help-template" data-title="{% trans "Label for Searching Again" %}"
-                    data-content="{% trans_html_attr "This text is for the search button displayed at the bottom of search results, so users can search again." %}">
-            </label>
-            <div class="{% css_field_class %}">
-              {% input_trans module.search_config.search_again_label.label langs input_name='search_again_label' input_id='search_again_label' data_bind="value: search_again_label" %}
+            {% if not request|toggle_enabled:'SPLIT_SCREEN_CASE_SEARCH' %}
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:158]: Nice attention to detail. If this is hidden, we also shouldn't show this text in the translations download: https://github.com/dimagi/commcare-hq/blob/b616afe421992ccc01293154d7e8a75e34eeb137/corehq/apps/translations/app_translations/download.py#L225-L226

File: corehq/apps/app_manager/tests/test_suite.py [hunk 4]
```diff
@@ -77,6 +79,46 @@ def test_case_search_action(self):
...
+                label={'en': 'Search Again'},
+            ),
+            properties=[CaseSearchProperty(name='name', label={'en': 'Name'})],
+        )
+        form_id = app.modules[0].forms[0].unique_id
+        app.modules[0].case_list_form = CaseListForm(
+            form_id=form_id,
+            label={'en': 'Registration Form'},
+        )
+        app = Application.wrap(app.to_json())
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_suite.py:99]: Why is this line needed?

File: corehq/apps/app_manager/tests/test_suite.py [hunk 5]
```diff
@@ -77,6 +79,46 @@ def test_case_search_action(self):
...
+
+        # case list actions unaffected
+        self.assertXmlHasXpath(
+            app.create_suite(),
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_suite.py:103]: This test should run a little faster if you just generate the suite once.

File: corehq/apps/app_manager/suite_xml/sections/details.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/suite_xml/sections/details.py:238]: nevermind, you were merging while I was reviewing

Comment (orangejenny) [reply corehq/apps/app_manager/suite_xml/sections/details.py:238]: Looking at the new diff, should [this new call to get_case_search_action](https://github.com/dimagi/commcare-hq/blob/b616afe421992ccc01293154d7e8a75e34eeb137/corehq/apps/app_manager/suite_xml/sections/details.py#L687-L694) also check the split screen flag?

Comment (nospame) [reply corehq/apps/app_manager/suite_xml/sections/details.py:238]: I saw that and wasn't immediately sure, but yes I think it should.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html (hunk 3 above)
Comment (nospame) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:158]: That makes sense. Does it matter what order those rows in translation downloads are in? Thinking it might be cleaner to do something like 
```
rows = [rows that are always present]
if not toggles.SPLIT_SCREEN_CASE_SEARCH.enabled(domain):
    rows.append(search_again_label)
```
rather than adding a check for it in the middle, but it could be less nice for a user if we're trying to match order to UI.

Comment (orangejenny) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:158]: I don't think the order in the download file matters, I'd go for the cleaner code

File: corehq/apps/app_manager/tests/test_suite.py (hunk 4 above)
Comment (nospame) [reply corehq/apps/app_manager/tests/test_suite.py:99]: I used the setup structure from the [test above it](https://github.com/dimagi/commcare-hq/blob/5f7be3abcde31b37285e1ca0c08f7557a3780593/corehq/apps/app_manager/tests/test_suite.py#L38-L53) but didn't include the comment: `# wrap to have assign_references called`

Comment (orangejenny) [reply corehq/apps/app_manager/tests/test_suite.py:99]: I'd vote for just calling `app.modules[0].assign_references()` then, but this isn't a blocking comment.

File: corehq/apps/translations/tests/test_bulk_app_translation.py [hunk 6]
```diff
@@ -1085,7 +1085,16 @@ def test_module_search_labels_rows(self):
...
         self.assertEqual(get_module_search_command_rows(app.langs, app.modules[0], app.domain),
                          [('search_label', 'list', 'Find a Mother'),
-                          ('search_again_label', 'list', 'Find Another Mother'),
+                          ('title_label', 'list', 'Find a Mom'),
+                          ('description', 'list', 'More information'),
+                          ('search_again_label', 'list', 'Find Another Mother')])
+
+    @flag_enabled('USH_CASE_CLAIM_UPDATES')
+    @flag_enabled('SPLIT_SCREEN_CASE_SEARCH')
+    def test_module_split_screen_case_search_rows(self):
```
Comment (orangejenny) [inline corehq/apps/translations/tests/test_bulk_app_translation.py:1094]: nice

**Review by orangejenny — APPROVED** [review]
